### ISSUE TITLE
Something do not exist?

### ISSUE THREAD

**Issue by RayZhangSB** [yshean/abnormal-spatiotemporal-ae#1]
I used the 'avenue' dataset to train the model ,when I test the model ,the file 'share/data/groundtruths/gt_{0}_vid{1:02d}.txt' can't open,and I didn't find the code generating this file. Please tell me the source of this file ,thanks.

**Comment by yshean** [comment]
Hi, sorry I missed your message. I'm using Hasan's groundtruth files (Google drive link [here](https://drive.google.com/drive/u/0/folders/0B7gqi2vg-71pRjFoV3FiMEc1cHM)) to run the evaluation. By the way, the codes in this repo are outdated and not in sync with the latest version. Will get it updated as soon as I finished the work on hand.

**Comment by KingMV** [comment]
@yshean  Hello,thanks for your code，your work help me very much! Have you finish your the newest code?when will you update your code？

**Comment by wyl0608** [comment]
@RayZhangSB  How do you solve the problem about `share/data/groundtruths/gt_{0}_vid{1:02d}.txt` does not exist?

**Comment by cbasavaraj** [comment]
Hey, I managed to get the Matlab .mat file from the Google drive link and make .txt files for each test video. If anyone's interested, uploading here.
[testing_gts.zip](https://github.com/yshean/abnormal-spatiotemporal-ae/files/2173877/testing_gts.zip)

**Comment by cbasavaraj** [comment]
Also, I had issues running training because the HDF5 file prepared after preprocessing was huge (23 GB). So I had to figure out how to use model.fit_generator(..) and managed to do that as well.

I can issue a pull request. But the authors have said previously that this repo is not up to date. So not sure how useful it would be.

If anyone's interested, let me know.

**Comment by cbasavaraj** [comment]
Hi @sebad78, it's been a while and eventually I just went back to image classification rather than using the video. But here's the relevant section of the code:

`    logger.info("Preparing training and testing data")
    preprocess_data(logger, dataset, time_length, video_root_path)
    hdf5file = os.path.join(video_root_path, '{0}/{0}_train_t{1}.h5'.format(dataset, time_length))
    hdf5data = HDF5Matrix(hdf5file, 'data')

    logger.info("Setting up training and validation data generators")

    all_size = hdf5data.end
    val_ratio = 0.2
    trn_size = int((1 - val_ratio) * all_size)
    val_size = all_size - trn_size

    print(all_size, trn_size, val_size)

    trn_data = HDF5Matrix(hdf5file, 'data', end=trn_size)
    val_data = HDF5Matrix(hdf5file, 'data', start=trn_size)

    print(trn_data.shape, val_data.shape)

    def generator(data, batch_size):
        size = data.shape[0]
        idx = 0
        while True:
            end = idx + batch_size
            if end > size:
                end = size
            yield data[idx:end], data[idx:end]
            idx = end % size

    trn_gen = generator(trn_data, batch_size)
    val_gen = generator(val_data, batch_size)

    trn_steps = trn_size // batch_size
    if trn_size % batch_size:
        trn_steps += 1

    val_steps = val_size // batch_size
    if val_size % batch_size:
        val_steps += 1

    logger.info("Initializing training...")

    snapshot = ModelCheckpoint(os.path.join(job_folder,
                                            'model_snapshot_e{epoch:03d}_{val_loss:.6f}.h5'))
    earlystop = EarlyStopping(patience=10)
    history_log = LossHistory(job_folder=job_folder, logger=logger)

    callbacks = [snapshot, earlystop, history_log]

    history = model.fit_generator(generator=trn_gen,
                                  steps_per_epoch=trn_steps,
                                  epochs=nb_epochs,
                                  verbose=2,
                                  validation_data=val_gen,
                                  validation_steps=val_steps,
                                  callbacks=callbacks)

    #   shuffle='batch',

    logger.info("Training completed!")
    np.save(os.path.join(job_folder, 'train_profile.npy'), history.history)`

**Comment by sebad78** [comment]
Hi @cbasavaraj Thank you very much!! It worked, I was creating a generator, but it was bad implemented I guess, because it throws OOM... 
Now its working with your changes. 
Best Regards.

**Comment by Abudhagir** [comment]
Hi @cbasavaraj I am unable to load the model for testing. Can u help me out.

  File "E:/AbnormalYShean/abnormal-spatiotemporal-ae-master/start_test.py", line 55, in <module>
    visualize_score=True, visualize_frame=False)
  File "E:\AbnormalYShean\abnormal-spatiotemporal-ae-master\classifier.py", line 235, in test
    temporal_model = load_model(os.path.join(job_folder, model_filename))

**Comment by viethungtsdv** [comment]
hi @yshean 
Could you please share the last version of source code? Thank you very much.

**Comment by cbasavaraj** [comment]
I just got an email from another person regarding the code. Don't have the time to issue a pull request right now, but attaching my version here. 

[abnormal-spatiotemporal-ae.zip](https://github.com/yshean/abnormal-spatiotemporal-ae/files/2755075/abnormal-spatiotemporal-ae.zip)

Best,
Chan
